scenario/constraints/MigrationConstraint.ts

```diff
@@ -1,17 +1,18 @@
 import { Constraint, Scenario, Solution, World } from '../../plugins/scenario';
 import { CometContext } from '../context/CometContext';
-import { ProtocolConfiguration, deployComet } from '../../src/deploy';
-import { getFuzzedRequirements } from './Fuzzing';
-import CometAsset from '../context/CometAsset';
 import { Contract } from 'ethers';
 import { Requirements } from './Requirements';
-
-import * as path from 'path';
 import { Migration, loadMigrations } from '../../plugins/deployment_manager/Migration';
 
-// TODO: Improvements
-function getMigrationConfigs(requirements: Requirements): boolean {
-  return requirements.migrate ?? false;
+import { execSync } from 'child_process';
+import { existsSync } from 'fs';
+
+async function getMigrations<T>(context: CometContext, requirements: Requirements): Promise<Migration<T>[]> {
+  // TODO: make this configurable from cli params/env var?
+  const deployment = context.deploymentManager.deployment; // XXX should become per instance
+  const output = execSync(`git diff --numstat main | grep 'deployments/${deployment}/{deploys,changes}/.*.ts' | awk '{ print $3 }'`);
```

Comment:
I wonder if there's a native JavaScript git library as opposed to running exec here?

Reply:
There's probably `libgit2` bindings but its another native dependency and API to figure out and not sure I see the gain right now

Reply:
yeah nodegit is too native, there is isogit and a complex process to get a diff without any native system calls: https://isomorphic-git.org/docs/en/snippets#git-diff-name-status-commithash1-commithash2

Reply:
`statusMatrix` seems closest to what we'd want, but would be a bunch of effort to match what its doing now fairly trivially, so will leave it as a future exercise

Reply:
Agreed.